.github/workflows/full-build.yml

```diff
@@ -143,7 +143,7 @@ jobs:
   macos:
     name: macOS
     needs: lint
-    runs-on: macos-11
+    runs-on: macos-latest
```

Comment:
What's the reason we cannot stay on macos-11? I'd rather have a fixed version anyway, so we can communicate that on our website.

Reply:
Hey @amolenaar, macos-12 has been available as a runner for over a year and homebrew is printing lots of warnings:

```
Warning: You are using macOS 11.
We (and Apple) do not provide support for this old version.
It is expected behaviour that some formulae will fail to build in this old version.
It is expected behaviour that Homebrew will be buggy and slow.
Do not create any issues about this on Homebrew's GitHub repositories.
Do not create any issues even if you think this message is unrelated.
Any opened issues will be immediately closed without response.
Do not ask for help from Homebrew or its maintainers on social media.
You may ask for help in Homebrew's discussions but are unlikely to receive a response.
Try to figure out the problem yourself and submit a fix as a pull request.
We will review it but may or may not accept it.
```

I was having issues with homebrew pulling in old versions. I think the brew update was what actually resolved this, but I think we should probably upgrade anyway. How about I pin macos-12?